### PR TITLE
V2

### PR DESCRIPTION
**Pull request by matthew-andrews**
~~ do not merge ~~

@JakeChampion would you mind taking this for a spin in one of your apps

Todo before merging:-
- [x] switch back to prod environment
- [ ] wait for mr barrows to put the latest version of V2 in prod

### CODE REVIEW

**Comment by matthew-andrews** [comment]
I fixed a bit of whitespace dissidence, so best to view diff like this: https://github.com/Financial-Times/s3o-middleware/pull/9/files?w=false

File: index.js [hunk 1]
```diff
@@ -53,49 +54,55 @@ var authS3O = function(req, res, next) {
...
 	}
 	// Check for s3o username/token cookies
 	else if (req.cookies.s3o_username && req.cookies.s3o_token) {
 		debug("S3O: Found cookie token for s3o_username: " + req.cookies.s3o_username);
 
-		if (authenticateToken(res, req.cookies.s3o_username, req.cookies.s3o_token)) {
+		if (authenticateToken(res, req.cookies.s3o_username, req.hostname, req.cookies.s3o_token)) {
 			next();
+		} else {
+			res.send("<h1>Authentication error.</h1><p>For access, please login with your FT account</p>");
```
Comment (matthew-andrews) [inline index.js]: got a bit confused when the helper function actually sent the headers…  this way the helper function is a bit ‘safer’… in a later PR I plan to abstract the error sending into a single method that all the errors that this middleware generates can be output

**Comment by JakeChampion** [comment]
Tested on [Screens](https://github.com/ftlabs/Screens) and it works as expected

File: index.js (hunk 1 above)
Comment (JakeChampion) [inline index.js]: Sounds good to me :+1:

**Pull request merged**